sched/scheduler/scheduler.go

```diff
@@ -8,6 +10,10 @@ import (
 	"sync"
 )
 
+type Scheduler interface {
```

Comment:
What if Scheduler just took a chan WorkItem and pulled from it? That might be a clearer interface and also be easier to implement...